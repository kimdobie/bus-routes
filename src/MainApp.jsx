/* eslint-disable react/prop-types */
import React from 'react';

import { BrowserRouter, Route, Switch } from 'react-router-dom';
import App from './components/App';

const MainApp = () => {
  return (
    <BrowserRouter>
      <div className='app'>
        <Switch>
          <Route exact path='/' component={App} />
          <Route
            path='/route/:route/direction/:direction'
            component={props => {
              return (
                <App
                  route={props.match.params.route}
                  direction={props.match.params.direction}
                />
              );
            }}
          />
          <Route
            path='/route/:route'
            component={props => {
              return <App route={props.match.params.route} />;
            }}
          />
        </Switch>
      </div>
    </BrowserRouter>
  );
};

export default MainApp;
