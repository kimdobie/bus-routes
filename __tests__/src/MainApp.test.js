import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import MainApp from '../../src/MainApp';

configure({ adapter: new Adapter() });

jest.mock('axios');

describe('Main App tests', () => {
  test('Snapshot test', async () => {
    // eslint-disable-next-line react/jsx-filename-extension
    const index = await shallow(<MainApp />);

    expect(index).toMatchSnapshot();
  });
});
// See https://medium.com/@antonybudianto/react-router-testing-with-jest-and-enzyme-17294fefd303
