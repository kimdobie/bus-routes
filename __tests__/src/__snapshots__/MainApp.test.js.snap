// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main App tests Snapshot test 1`] = `
<BrowserRouter>
  <div
    className="app"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/route/:route/direction/:direction"
      />
      <Route
        component={[Function]}
        path="/route/:route"
      />
    </Switch>
  </div>
</BrowserRouter>
`;
